src/mpl/src/gpu/mpl_gpu_cuda.c

```diff
@@ -4,10 +4,22 @@
  */
 
 #include "mpl.h"
+#include <assert.h>
 
 #define CUDA_ERR_CHECK(ret) if (unlikely((ret) != cudaSuccess)) goto fn_fail
 #define CU_ERR_CHECK(ret) if (unlikely((ret) != CUDA_SUCCESS)) goto fn_fail
 
+typedef struct {
+    uintptr_t remote_base_addr;
+    uintptr_t dev_addr;
```

Comment:
Is `dev_addr` supposed to be the local mapped address?  If yes, I'd recommend naming it as `mapped_addr`.